ecdsa/src/rfc6979.rs

```diff
@@ -28,82 +29,47 @@ where
     Scalar<C>:
         PrimeField<Repr = FieldBytes<C>> + FromDigest<C> + Invert<Output = Scalar<C>> + Zeroize,
 {
-    let mut x = secret_scalar.to_repr();
-    let h1 = Scalar::<C>::from_digest(msg_digest).to_repr();
-    let mut hmac_drbg = HmacDrbg::<D>::new(&x, &h1, additional_data);
+    let bits_to_bytes = |b: elliptic_curve::ScalarBits<C>| {
+        // make byte buffer of the right size
+        let mut tmp = FieldBytes::<C>::default();
+        // convert bitslice from internal repr based on u32 or u64 to one based on u8
+        tmp.as_mut_slice()
+            .view_bits_mut::<Lsb0>()
+            .clone_from_bitslice(b.as_bitslice());
+        // convert to big endian
+        tmp.reverse();
+        tmp
+    };
+
+    // Verify that values roundtrip through bits_to_bytes and from_repr as expected.
+    // The from_repr interface does not guarantee big endian interpretation, and if it is not
+    // as expected the generated value may not have the required entropy.
+    let one = Scalar::<C>::one();
+    let one_bytes = bits_to_bytes(one.to_le_bits());
+    assert!(
+        one_bytes[one_bytes.len() - 1] == 1 && Scalar::<C>::from_repr(one_bytes) == Some(one),
+        "curve repr not as expected"
+    );
+
+    // Convert inputs to big endian byte representation
+    let modulus = bits_to_bytes(Scalar::<C>::char_le_bits());
+    let h1 = bits_to_bytes(Scalar::<C>::from_digest(msg_digest).to_le_bits());
```

Comment:
It might be nicer to pass the `Digest` down all the way to the new crate instead of converting everything to byte buffers here. Unfortunately the new crate can't use the `FromDigest` trait, because it's declared in `ecdsa` and depending on `ecdsa` would create a circular dependency between the `ecdsa` and `rfc6079` crates.

Reply:
If it's helpful to push the `FromDigest` trait down to the `rfc6979` crate, I guess I'd be ok with having it there.

[As I mentioned earlier](https://github.com/RustCrypto/signatures/pull/244#issuecomment-778550307) in ECDSA the specific transformation is a core part of the algorithm and not specifically related to RFC6979, but what ECDSA does and what RFC6979 needs are the same, so I guess I wouldn't be opposed to moving it.